### PR TITLE
fix: step status weight settings; enhanced tooltip

### PR DESCRIPTION
**Pull request by khpeet**
- Step weight configuration - Allows to include/exclude signals from step weight calculation, as well as signal rollup options
- Enhanced step tooltip - Includes quick detail in step status config and a drilldown into currently selected signals contributing to step status determination
- Added status outline to `Inline` view of step cells

### CODE REVIEW

**Comment by khpeet** [comment]
fixes #439

File: src/components/step-settings-modal/index.js [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+  useEffect(() => {
+    if (
+      stepStatusValue !== '' &&
+      stepStatusOption &&
+      stepStatusOption === STEP_STATUS_OPTIONS.WORST
+    ) {
+      setStatusWeightValue(stepStatusValue);
+      setChecked(true);
+    }
+  }, [stepStatusValue]);
```
Comment (amit-y) [inline src/components/step-settings-modal/index.js:100]: Add `stepStatusOption` as a dependency

File: src/components/step-settings-modal/index.js [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+      setStatusWeightUnit(STEP_STATUS_UNITS.PERCENT);
+      setStatusWeightValue('');
+      setChecked(false);
+    }
+    setStatusOption(value);
+  }, []);
+
+  const toggleApplySwitch = useCallback(() => {
+    setChecked((prevChecked) => !prevChecked);
+  }, [signals]);
```
Comment (amit-y) [inline src/components/step-settings-modal/index.js:167]: Don't need the `signals` dependency

File: src/components/step-tooltip/index.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    const [viewRuleModalHidden, setViewRuleModalHidden] = useState(true);
+
+    useEffect(() => {
+      if (signals.length > 0) {
+        const includedSignals = signals.filter((s) => {
+          return s.included === true || s.included === undefined;
+        });
+
+        setIncludedSignals(includedSignals);
+      }
```
Comment (amit-y) [inline src/components/step-tooltip/index.js:35]: I would change this to...
```javascript
if (signals.length > 0) {
  setIncludedSignals(() => signals.filter((s) => {
    return s.included === true || s.included === undefined;
  }));
```

`includedSignals` is already defined as a state variable, and to remove ambiguity, I'd try not creating a local variable named `includedSignals`

File: src/components/step-tooltip/styles.scss [hunk 4]
```diff
@@ -0,0 +1,154 @@
+.StepTooltip {
```
Comment (amit-y) [inline src/components/step-tooltip/styles.scss:1]: Ideally use the naming style we use in the rest of the project. For class names, use kebab-case. So this one would be `step-tooltip`

File: src/components/step-tooltip/styles.scss [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+  background-color: #202d32;
+  box-shadow: rgba(#283636, 0.15) 16px 8px 32px;
+  box-sizing: border-box;
+  padding: 4px 0;
+  //width: 324px;
```
Comment (amit-y) [inline src/components/step-tooltip/styles.scss:6]: Please remove comments.

File: nerdlets/signal-selection/index.js [hunk 6]
```diff
@@ -219,15 +219,17 @@ const SignalSelectionNerdlet = () => {
           levelId,
           stepId: step?.id,
           signals: [
-            ...(selectedEntities || []).map(({ guid, name }) => ({
+            ...(selectedEntities || []).map(({ guid, name, included }) => ({
               guid,
               name,
               type: SIGNAL_TYPES.ENTITY,
+              included: included !== undefined ? included : true,
```
Comment (amit-y) [inline nerdlets/signal-selection/index.js:226]: Not sure what the exact logic here, but if you're just checking to see is `included` is truthy, it's simpler to do this...
```javascript
included: included || true,
```

Comment (khpeet) [reply nerdlets/signal-selection/index.js:226]: I originally had that and it didnt work - I'm checking if `included` exists in the object.

File: src/components/step-tooltip/styles.scss [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+.step-tooltip-header {
+  border-bottom: #e7e9ea 1px solid;
+  align-items: baseline;
+  margin-left: 8px;
+  margin-right: 8px;
+  padding-bottom: calc(8px);
+  padding-left: 8px;
+  display: flex;
+
+  &-statusIndicator {
```
Comment (amit-y) [inline src/components/step-tooltip/styles.scss:20]: Kebab-case, please

File: src/components/step-tooltip/styles.scss [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+  &.header {
+    margin-bottom: 4px;
+  }
+
+  span {
+    margin-bottom: 8px;
+  }
+
+  &.inclusionList {
```
Comment (amit-y) [inline src/components/step-tooltip/styles.scss:99]: Kebab-case please

File: src/components/step-tooltip/styles.scss [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  &-overview {
+    &::after {
+      border-right: 1px solid;
+      content: '';
+      height: 100%;
+      margin-left: 4px;
+    }
+  }
+
+  &-viewRule {
```
Comment (amit-y) [inline src/components/step-tooltip/styles.scss:76]: Kebab-case please

**Review by amit-y — APPROVED** [review]

**Pull request merged**